jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/JunitTestsPublisher.java

```diff
@@ -334,8 +335,7 @@ private void executeReporter(StepContext context, TaskListener listener, List<El
                 if (LOGGER.isLoggable(Level.FINE)) {
                     listener.getLogger().println("[withMaven] junitPublisher - " + testResultAction.getResult().getFailCount() + " unit test failure(s) found, mark job as unstable");
                 }
-                // TODO: Once JENKINS-43995 lands, update this to set the step status instead of the entire build.
-                // context.setResult(Result.UNSTABLE);
+                node.addAction(new WarningAction(Result.UNSTABLE).withMessage(testResultAction.getResult().getFailCount() + " unit test failure(s) found"));
```

Comment:
Note: I _suppose_ multiple `WarningActions` per node are ok (the API seems to permit it) but I haven't checked what happens when multiple such actions are present.
As comparison, `org.jenkinsci.plugins.workflow.steps.UnstableStep` is using `addOrReplaceAction()` instead of `addAction()` but I don't think it makes sense to overwrite other warnings here...